moit-domain/src/main/kotlin/com/mashup/moit/domain/attendance/AttendanceService.kt

```diff
@@ -2,14 +2,54 @@ package com.mashup.moit.domain.attendance
 
 import com.mashup.moit.common.exception.MoitException
 import com.mashup.moit.common.exception.MoitExceptionType
+import com.mashup.moit.domain.study.StudyEntity
+import com.mashup.moit.domain.study.StudyRepository
+import com.mashup.moit.domain.usermoit.UserMoitRepository
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDateTime
 
 @Service
 @Transactional(readOnly = true)
 class AttendanceService(
     private val attendanceRepository: AttendanceRepository,
+    private val studyRepository: StudyRepository,
+    private val userMoitRepository: UserMoitRepository,
 ) {
+    @Transactional
+    fun initializeAttendance(studyId: Long) {
+        val study = studyRepository.findById(studyId).orElseThrow { MoitException.of(MoitExceptionType.NOT_EXIST) }
+        val userMoits = userMoitRepository.findAllByMoitId(study.moitId)
+
+        userMoits.map {
+            AttendanceEntity(
+                studyId = studyId,
+                userId = it.userId,
+                status = AttendanceStatus.UNDECIDED,
+                attendanceAt = null,
+            )
+        }.let { attendanceRepository.saveAll(it) }
+    }
+
+    @Transactional
+    fun requestAttendance(userId: Long, studyId: Long): Attendance {
+        val study = studyRepository.findById(studyId).orElseThrow { MoitException.of(MoitExceptionType.NOT_EXIST) }
+        val attendance = attendanceRepository.findByUserIdAndStudyId(userId, studyId)
+            ?: throw MoitException.of(MoitExceptionType.ATTENDANCE_NOT_STARTED)
+        if (attendance.status !== AttendanceStatus.UNDECIDED) {
+            throw MoitException.of(
+                MoitExceptionType.ALREADY_EXIST,
+                "이미 출석하였습니다."
+            )
+        }
+
+        val now = LocalDateTime.now()
+        return attendance.apply {
+            this.status = study.attendanceStatus(now)
+            this.attendanceAt = now
+        }.toDomain()
```

Comment:
`now` 따로 안빼고 inline으로 넣어버려도 될 것 같긴 합니다.
```suggestion
        return attendance.apply {
            this.status = study.attendanceStatus(now)
            this.attendanceAt = LocalDateTime.now()
        }.toDomain()
```

Reply:
음 AttendanceStatus 를 계산할 때 사용하는 기준 시간이랑, attendanceAt 은 같은 now 를 사용하는게 맞지 않을까요..?
혹시 제가 놓치고 있는 부분이 있다면 말씀 부탁드립니다..! 🙇